### PR TITLE
Extend register sign using VASM for PPC64

### PR DESCRIPTION
**Pull request by lbianc**
The VASMs ```loadw``` and ```loadl``` was implicity extending the sign when
emitting the instruction. It was creating invalid VASM sequences, as right
after a ```loadw``` may appear a ```cmpq```, which uses different register sizes.
Another inconsistency was the ```loadb``` VASM instruction, which for ```PPC64``` by
default is not signed. When loading a byte and comparing with another
value, the invalid VASM sequence ```loadb```; ```cmpq``` was being created. It was
fixed lowering the ```loadb``` instruction to ```loadzbq```, in order to explicitly
extend the size.

### CODE REVIEW

**Comment by lbianc** [comment]
@mxw

File: hphp/runtime/vm/jit/vasm-ppc64.cpp [hunk 1]
```diff
@@ -1022,13 +1013,27 @@ void lowerForPPC64(const VLS& e, Vout& v, vasm_src& inst) {             \
...
+#define X(vasm_src, vasm_dst_reg, vasm_dst_imm, vasm_load, vasm_exts)   \
+void lowerForPPC64(const VLS& e, Vout& v, vasm_src& inst) {             \
+  Vptr p = inst.s1;                                                     \
+  Vreg tmp; if (!e.allow_vreg()) tmp = Vreg(PhysReg(rAsm));             \
+  Vreg tmp2 = e.allow_vreg() ? v.makeReg() : Vreg(PhysReg(rAsm));       \
+  Vreg tmp3 = v.makeReg();                                              \
+  v << vasm_load{p, tmp2};                                              \
+  v << vasm_exts{tmp2, tmp3};                                           \
+  if (patchImm(inst.s0, v, tmp))                                        \
+    v << vasm_dst_reg{tmp, tmp3, inst.sf};                              \
```
Comment (mxw) [inline hphp/runtime/vm/jit/vasm-ppc64.cpp]: Braces are required for multiline conditionals.

File: hphp/runtime/vm/jit/vasm-ppc64.cpp [hunk 2]
```diff
@@ -1022,13 +1013,27 @@ void lowerForPPC64(const VLS& e, Vout& v, vasm_src& inst) {             \
...
-X(testwim, testq, testqi, loadw)
-X(testlim, testq, testqi, loadl)
+X(testbim, testq, testqi, loadzbq)
 X(testqim, testq, testqi, load)
+#undef X
+
+#define X(vasm_src, vasm_dst_reg, vasm_dst_imm, vasm_load, vasm_exts)   \
+void lowerForPPC64(const VLS& e, Vout& v, vasm_src& inst) {             \
+  Vptr p = inst.s1;                                                     \
+  Vreg tmp; if (!e.allow_vreg()) tmp = Vreg(PhysReg(rAsm));             \
```
Comment (mxw) [inline hphp/runtime/vm/jit/vasm-ppc64.cpp]: Rewrite this as a ternary.

Comment (lbianc) [reply hphp/runtime/vm/jit/vasm-ppc64.cpp]: I have removed the tmp set, as it is already made in ```patchImm``` function.

File: hphp/runtime/vm/jit/vasm-ppc64.cpp (hunk 1 above)
Comment (lbianc) [reply hphp/runtime/vm/jit/vasm-ppc64.cpp]: Done.

**Pull request closed without merging**